plugins/github-actions/src/plugin.ts

```diff
@@ -23,10 +23,18 @@ import {
   githubAuthApiRef,
   createRoutableExtension,
   createComponentExtension,
+  createComponentAdaptationExtension,
 } from '@backstage/core-plugin-api';
+import { linkComponentRef } from '@backstage/core-components';
 
 export const githubActionsPlugin = createPlugin({
   id: 'github-actions',
+  adaptations: {
+    link: createComponentAdaptationExtension(linkComponentRef, {
```

Comment:
Could call this `createComponentAdaptation` right?

Reply:
Ah right didn't realize this was actually an extension as well, should make sense if we structure it like that. I'm not entirely sure this should be an extension though, but perhaps something we can come back to after settling other pieces first.

Reply:
Yeah, the benefit of extensionalizing adaptations is the its logic can be wrapped in / tied to the plugin that provides it, so that errors thrown within an adaptation (which is "ok", or rather "handled" - that adaptation step will be dropped) can be bound to the plugin for ownership / debugging purposes.